app/models/vm_or_template.rb

```diff
@@ -228,15 +228,17 @@ def from_infra_manager?
 
   # Add virtual columns/methods for specific things derived from advanced_settings
   REQUIRED_ADVANCED_SETTINGS = {
-    'vmi.present'         => [:paravirtualization,   :boolean],
-    'vmsafe.enable'       => [:vmsafe_enable,        :boolean],
-    'vmsafe.agentAddress' => [:vmsafe_agent_address, :string],
-    'vmsafe.agentPort'    => [:vmsafe_agent_port,    :integer],
-    'vmsafe.failOpen'     => [:vmsafe_fail_open,     :boolean],
-    'vmsafe.immutableVM'  => [:vmsafe_immutable_vm,  :boolean],
-    'vmsafe.timeoutMS'    => [:vmsafe_timeout_ms,    :integer],
-    'entitled_processors' => [:entitled_processors,  :float],
-    'processor_type'      => [:processor_share_type, :string],
+    'vmi.present'          => [:paravirtualization,   :boolean],
+    'vmsafe.enable'        => [:vmsafe_enable,        :boolean],
+    'vmsafe.agentAddress'  => [:vmsafe_agent_address, :string],
+    'vmsafe.agentPort'     => [:vmsafe_agent_port,    :integer],
+    'vmsafe.failOpen'      => [:vmsafe_fail_open,     :boolean],
+    'vmsafe.immutableVM'   => [:vmsafe_immutable_vm,  :boolean],
+    'vmsafe.timeoutMS'     => [:vmsafe_timeout_ms,    :integer],
+    'entitled_processors'  => [:entitled_processors,  :float],
+    'processor_type'       => [:processor_share_type, :string],
+    'processor_pin_policy' => [:pin_policy,           :string],
```

Comment:
I think this is the other way around, the first key is the name of the advanced setting from the provider and `:pin_policy` in this case is the name of the method defined on the VmOrTemplate model.  I was suggesting we rename the method defined on the model not the advanced setting since that's pre-defined by the provider.

Reply:
My bad. Let me reverse and make corrections.